modules/hawkular-wildfly-agent-download/src/main/java/org/hawkular/component/wildflymonitor/WildFlyAgentServlet.java

```diff
@@ -171,23 +273,42 @@ private void copy(InputStream input, OutputStream output) throws IOException {
         output.flush();
     }
 
-    private File getAgentDownloadFile() throws Exception {
-        if (downloadFile != null) {
-            if (downloadFile.exists()) {
-                return downloadFile;
+    private File getAgentModuleDownloadFile() throws Exception {
+        if (moduleDownloadFile != null) {
+            if (moduleDownloadFile.exists()) {
+                return moduleDownloadFile;
             } else {
-                downloadFile = null; // the file was removed recently - let's look for a new one
+                moduleDownloadFile = null; // the file was removed recently - let's look for a new one
             }
         }
 
         File configDir = new File(System.getProperty("jboss.server.config.dir"));
         for (File file : configDir.listFiles()) {
             if (file.getName().startsWith("hawkular-wildfly-agent-wf-extension") && file.getName().endsWith(".zip")) {
-                downloadFile = file;
-                return downloadFile;
+                moduleDownloadFile = file;
+                return moduleDownloadFile;
+            }
+        }
+        throw new FileNotFoundException("Cannot find agent module download in: " + configDir);
+    }
+
+    private File getAgentInstallerDownloadFile() throws Exception {
+        if (installerDownloadFile != null) {
+            if (installerDownloadFile.exists()) {
+                return installerDownloadFile;
+            } else {
+                installerDownloadFile = null; // the file was removed recently - let's look for a new one
+            }
+        }
+
+        File configDir = new File(System.getProperty("jboss.server.config.dir"));
+        for (File file : configDir.listFiles()) {
+            if (file.getName().startsWith("agent-installer") && file.getName().endsWith(".jar")) {
```

Comment:
we want to continue using the official branding 'hawkular-wildfly-agent' not just 'agent'  (because in the world of hawkular, there are many kinds of feeds and agents, hawkular-wildfly-agent being just one flavor.)

So we need to make sure the names of the files are reflective of the hawkular name - "hawkular-wildfly-agent-installer-ABC.jar"